src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAjaxController.java

```diff
@@ -1148,6 +1149,36 @@ public AnalysisProvenanceResponse getProvenanceByFile(@PathVariable Long submiss
 		return analysisProvenance;
 	}
 
+	/**
+	 * Get the updated state and duration of an analysis
+	 *
+	 * @param submissionId The analysis submission id
+	 * @return dto which contains the updated analysis state and duration
+	 */
+	@RequestMapping(value = "/{submissionId}/updated-progress")
+	@ResponseBody
+	public UpdatedAnalysisProgress getProvenanceByFile(@PathVariable Long submissionId) {
```

Comment:
Wrap this in a `ResponseEntity` so you can return an error state if you need to.

Reply:
Updated in c64a856128d212fdda1bcab36dbf56b78cef8399